File: frontend/src/app/services/orden-repuesto.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { OrdenRepuesto } from '../model/ordenRepuesto';

export interface OrdenRepuestoRequest {
  ordenId: number;
  repuestoId: number;
  cantidad: number;
  costoUnitario: number;
}

@Injectable({
  providedIn: 'root'
})
export class OrdenRepuestoService {

  private apiUrl = environment.apiUrl;
  private http = inject(HttpClient);

  // 🔥 agrega un repuesto/fungible a una orden YA EXISTENTE, guardando de
  // inmediato en la BD (descuenta stock al instante, sin esperar al
  // botón "Actualizar" que ya no existe para este flujo).
  agregar(request: OrdenRepuestoRequest): Observable<OrdenRepuesto> {
    return this.http.post<OrdenRepuesto>(`${this.apiUrl}/orden-repuestos`, request);
  }

  // 🔥 elimina un repuesto ya guardado, reponiendo el stock que se
  // había descontado al agregarlo.
  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/orden-repuestos/${id}`);
  }

  listarPorOrden(ordenId: number): Observable<OrdenRepuesto[]> {
    return this.http.get<OrdenRepuesto[]>(`${this.apiUrl}/orden-repuestos/orden/${ordenId}`);
  }
}

File: svuaback/src/main/java/cl/aracridav/svua/mantenimiento/ordenrepuesto/service/OrdenRepuestoService.java
package cl.aracridav.svua.mantenimiento.ordenrepuesto.service;

import java.util.List;

import cl.aracridav.svua.mantenimiento.ordenrepuesto.dto.request.OrdenRepuestoRequest;
import cl.aracridav.svua.mantenimiento.ordenrepuesto.dto.response.OrdenRepuestoResponse;

public interface OrdenRepuestoService {

    public OrdenRepuestoResponse agregarRepuesto(OrdenRepuestoRequest request);
    public List<OrdenRepuestoResponse> listarPorOrden(Long ordenId);

    // 🔥 elimina un repuesto ya asociado a una orden, reponiendo el
    // stock descontado al agregarlo.
    public void eliminarRepuesto(Long id);

}
